source/dub/compilers/ldc.d

```diff
@@ -259,6 +260,14 @@ config    /etc/ldc2.conf (x86_64-pc-linux-gnu)
 				break;
 			case TargetType.object:
 				settings.addDFlags("-c");
+
+				// When using wasm-ld on output objects, we need to explicitly
+				// not strip dead symbols, otherwise we'll get a linker error.
+				// as wasm-ld only works on relocatable objects.
+				if (p.canFind("wasm")) {
+					settings.addDFlags("--disable-linker-strip-dead");
+					settings.addLFlags("-r");
```

Comment:
this doesn't quite seem like the right place to add the dflags per platform, since there is also no other such check here, but I haven't been able to check the code in other compiler wrappers here yet, perhaps consider just disregarding this comment

Reply:
This requirement only applies to the object target type. Couldn't find anywhere better to put it given that .o files have to be reloctable and can't be section gc'd. If these flags aren't there for the object type then you will be unable to use wasm-ld and/or emscripten to link your module with eg. Emscripten ports or other external libraries.